### PR TITLE
e2e: add codecov uploader configuration

### PR DESCRIPTION
**Pull request by fmuyassarov**
e2e: add codecov uploader configuration
Depends on https://github.com/kubernetes/test-infra/pull/29082.

### CODE REVIEW

File: Makefile
Comment (marquiz) [inline Makefile:14]: What report(?) Maybe `UPLOAD_COVERAGE_REPORT` or smth?

File: Makefile
Comment (fmuyassarov) [reply Makefile]: Good point. I agree. Moved those steps to infra in https://github.com/kubernetes/test-infra/pull/29101.

File: codecov
Comment (marquiz) [inline codecov:1]: We don't want to add the codecov binary to the repo. We want to update the test-infra script to download it (and upload the report)

**Review by marquiz — COMMENTED** [review]
Don't we want to update the `scripts/test-infra/verify.sh` script?

**Comment by marquiz** [comment]
> For?

Download the codecov binary and upload the report. What you had in Makefile in the earler version

**Comment by fmuyassarov** [comment]
> > For?
> 
> Download the codecov binary and upload the report. What you had in Makefile in the earler version

ah, I read your comment too fast in https://github.com/kubernetes-sigs/node-feature-discovery/pull/1095#discussion_r1142980708 and understood that we want it to be as part of the test infra Prow job definition. I just noticed when read your comment again :) So, currently it is done in https://github.com/kubernetes/test-infra/pull/29101. We can keep wither there or in verify.sh. I don't have any preferences here.

**Comment by marquiz** [comment]
> ah, I read your comment too fast in [#1095 (comment)](https://github.com/kubernetes-sigs/node-feature-discovery/pull/1095#discussion_r1142980708) and understood that we want it to be as part of the test infra Prow job definition. I just noticed when read your comment again :) So, currently it is done in [kubernetes/test-infra#29101](https://github.com/kubernetes/test-infra/pull/29101). We can keep wither there or in verify.sh. I don't have any preferences here.

Ach sorry, I didn't realize that. I think it would make sense to move the stuff from [kubernetes/test-infra#29101](https://github.com/kubernetes/test-infra/pull/29101) to `scripts/test-infra/verify.sh`. Makes it easier to read and maintain IMO, plus we already do a bunch of installation/setup steps in verify.sh so it is logical to keep all of that in one place. WDYT?

**Comment by ArangoGutierrez** [comment]
Where are we here?  Can we think of v0.13 for this? or push to v0.14?

**Comment by marquiz** [comment]
@fmuyassarov is on PTO. We can push this beyond v0.13, not a user facing feature

**Comment by fmuyassarov** [comment]
/hold 
until keys are created

File: scripts/test-infra/verify.sh
Comment (marquiz) [inline scripts/test-infra/verify.sh:30]: I'd suggest to move the codecov tool installation to the "Install deps" part above where other installation and setup steps are done, too

**Comment by fmuyassarov** [comment]
We have one more dependency https://github.com/kubernetes/test-infra/pull/29280 and then we are good to merge this one if LGTY.

**Review by marquiz — APPROVED** [review]
Thanks @fmuyassarov, to me looks good to be merged

**Comment by fmuyassarov** [comment]
Let's give it a try!
/unhold

**Pull request merged**
